### PR TITLE
fix: Some elasticsearch versions return a dict for hits.total

### PR DESCRIPTION
**Pull request by lutzkuen**
While setting up amundsen i noticed that our elasticsearch returns a dict like {value: "20", rel: "eq"} for hits.total and not an int. This leads to us not getting any search results. This PR includes the changes I made to the searchlibrary to get it to work.

### Summary of Changes

Check in elasticsearch.py whether hits.total is int or dict

### Tests

Tests to test_elasticsearch.py added:
 test_search_with_empty_result_dict_total
 test_search_with_one_table_result_dict_total

copy off existing tests, just that the hits.total is a dict

### Documentation

No changes to documentation

### CheckList

Make sure you have checked **all** steps below to ensure a timely review.

-   [ x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    -   In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
-   [x ] PR includes a summary of changes.
-   [ x] PR adds unit tests, updates existing unit tests, **OR** documents why no test additions or modifications are needed.
-   [ x] In case of new functionality, my PR adds documentation that describes how to use it.
    -   All the public functions and the classes in the PR contain docstrings that explain what it does
-   [ x] PR passes `make test`

### CODE REVIEW

File: .github/workflows/publish_release.yml [hunk 1]
```diff
@@ -41,3 +41,16 @@ jobs:
         with:
           user: __token__
           password: ${{ secrets.pypi_password }}
+  build-and-publish-docker-image:
```
Comment (mgorsk1) [inline .github/workflows/publish_release.yml:44]: is this related to the change ?

File: setup.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 from setuptools import find_packages, setup
 
-__version__ = '2.5.0'
+__version__ = '2.5.1'
```
Comment (mgorsk1) [inline setup.py:8]: I think no need to bump manually, automatic release action will take care of it

File: search_service/proxy/elasticsearch.py [hunk 3]
```diff
@@ -157,8 +157,11 @@ def _get_search_result(self, page_index: int,
                 results.append(model(**result))
             except Exception:
                 LOGGING.exception('The record doesnt contain specified field.')
-
-        return search_result_model(total_results=response.hits.total,
+        if isinstance(response.hits.total, dict):
```
Comment (mgorsk1) [inline search_service/proxy/elasticsearch.py:160]: might be useful to add a comment why this might happend. dict is what ES 7 returns, ES 6 returns int

**Review by mgorsk1 — COMMENTED** [review]
I think this fix is needed when someone uses ES 7+. It introduces implicit support for ES7 correcting just one of possibly many incompatibility issues - dependencies are pinned for ES 6 support. I think it's rather safe but will hive a chance for others to have a look.

cc @feng-tao

**Comment by dorianj** [comment]
We should try to land this in monorepo, tagging
